decode.go

```diff
@@ -929,14 +929,9 @@ func (d *Decoder) decodeValue(ctx context.Context, dst reflect.Value, src ast.No
 					dst.SetUint(uint64(i))
 					return nil
 				}
-			} else { // couldn't be parsed as float
-				return errTypeMismatch(valueType, reflect.TypeOf(v), src.GetToken())
 			}
-
-		default:
-			return errTypeMismatch(valueType, reflect.TypeOf(v), src.GetToken())
 		}
-		return errOverflow(valueType, fmt.Sprint(v))
+		return errTypeMismatch(valueType, reflect.TypeOf(v), src.GetToken())
```

Comment:
I need the overflow information, so could you modify the `errOverflow` function to pass `src.GetToken` ?

Reply:
Thanks for your review. I am going to modify the error as you suggested.

Reply:
I have changed the approach of this PR. Instead of using another error, I have refactored the `overflowError` to be a `PrettyPrinter` and `Formatter` like the other error types.

Please take another look and thanks again for your review and providing this library.